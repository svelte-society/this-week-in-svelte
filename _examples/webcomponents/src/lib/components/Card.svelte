<svelte:options
	customElement={{
		tag: 'twis-card',
		props: {
			complex: {
				type: 'Object',
				reflect: true
			}
		}
	}}
/>

<script lang="ts">
	import Button from './Button.svelte';

	export let title = '';
	export let complex: { number: number };
</script>

<section>
	<h1>{title}{complex?.number}</h1>
	<hr />
	<slot />
	<div>
		<Button on:click>Select</Button>
	</div>
</section>

<style>
	section {
		--padding: 2rem;
		box-shadow: 0 0 1rem 0 rgba(0 0 0 / 0.3);
		padding: var(--padding);
		font-family: sans-serif;
		border-radius: 0.5rem;
	}
	hr {
		margin-inline: calc(-1 * var(--padding));
		margin-block: var(--padding);
	}
	h1 {
		margin: 0;
	}
	div {
		margin-block-start: var(--padding);
	}
</style>
