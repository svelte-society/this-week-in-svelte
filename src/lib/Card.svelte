<script>
	/** @type {'purple' | 'green' | null} */
	export let kind = null
</script>

<div
	class="card"
	class:purple={kind === 'purple'}
	class:green={kind === 'green'}
>
	<slot />
</div>

<style>
	.card {
		padding: 1.5rem;
		border: 1px solid black;
	}
	.purple {
		background-color: purple;
		color: white;
	}
	.green {
		background-color: green;
		color: white;
	}
</style>
