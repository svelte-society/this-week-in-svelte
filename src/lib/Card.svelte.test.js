import { afterEach, describe, expect, it } from 'vitest'
import { cleanup, render } from '@testing-library/svelte'
import Card from './Card.svelte'

describe('Card.svelte', () => {
	afterEach(() => cleanup())

	it('mounts', () => {
		const { container } = render(Card)
		expect(container).toBeTruthy()
		expect(container.innerHTML).toMatchSnapshot()
	})

	it('renders purple', () => {
		// <Card kind="purple" />
		const { container } = render(Card, { kind: 'purple' })
		expect(container.innerHTML).toMatchSnapshot()
	})
})
