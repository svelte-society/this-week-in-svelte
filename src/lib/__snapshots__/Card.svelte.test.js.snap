// Vitest Snapshot v1

exports[`Card.svelte > mounts 1`] = `"<div><div class=\\"card svelte-1g8u0r7\\"></div></div>"`;
